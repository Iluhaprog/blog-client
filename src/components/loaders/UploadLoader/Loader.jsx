import React from 'react';
import PropTypes from 'prop-types';
import loader from '../../../assets/loader.svg';
import './loader.scss';

function Loader(props){
    return (
        <>
            <img 
                src={loader}
                width={props.width || 30}
                height={props.height || 30} 
                className={`loader ${props.visible ? 'loader_visible' : ''}`}
                alt='loading...'     
            />
        </>
    );
}

Loader.defaultProps = {
    width: 30,
    height: 30,
    visible: false,
};

Loader.propTypes = {
    width: PropTypes.number,
    height: PropTypes.number,
    visible: PropTypes.bool,
};

export default Loader;