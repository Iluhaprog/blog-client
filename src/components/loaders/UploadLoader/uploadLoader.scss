.loader {
    margin: 5px 15px;
    transition-duration: .01s;
}