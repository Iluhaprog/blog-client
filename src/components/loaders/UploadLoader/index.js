import UploadLoader from './UploadLoader';

export {
    UploadLoader,
};