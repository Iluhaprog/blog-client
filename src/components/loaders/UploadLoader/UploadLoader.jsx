import React from 'react';
import PropTypes from 'prop-types';
import loader from '../../../assets/loader.svg';
import { Loader } from '../Loader';
import './uploadLoader.scss';

function UploadLoader(props){
    return (
        <Loader visible={props.visible}>
            <img 
                src={loader}
                width={props.width || 30}
                height={props.height || 30}
                alt='loading...'     
            />
        </Loader>
    );
}

UploadLoader.defaultProps = {
    width: 30,
    height: 30,
    visible: false,
};

UploadLoader.propTypes = {
    width: PropTypes.number,
    height: PropTypes.number,
    visible: PropTypes.bool,
};

export default UploadLoader;