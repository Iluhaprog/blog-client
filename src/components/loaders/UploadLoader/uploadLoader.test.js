import React from 'react';
import { render } from '@testing-library/react';
import UploadLoader from './UploadLoader';

test('Should return loader view (visible false)', () => {
    const { asFragment } = render(<UploadLoader visible={false} />)
    expect(asFragment()).toMatchSnapshot();
});

test('Should return loader view (visible true)', () => {
    const { asFragment } = render(<UploadLoader visible={true} />)
    expect(asFragment()).toMatchSnapshot();  
});