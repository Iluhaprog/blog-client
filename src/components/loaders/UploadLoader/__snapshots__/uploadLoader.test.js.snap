// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should return loader view (visible false) 1`] = `
<DocumentFragment>
  <div
    class="loader "
  >
    <img
      alt="loading..."
      height="30"
      src="loader.svg"
      width="30"
    />
  </div>
</DocumentFragment>
`;

exports[`Should return loader view (visible true) 1`] = `
<DocumentFragment>
  <div
    class="loader loader_visible"
  >
    <img
      alt="loading..."
      height="30"
      src="loader.svg"
      width="30"
    />
  </div>
</DocumentFragment>
`;
